### PR TITLE
Comments on login

### PR DESCRIPTION
**Pull request by olu144**

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -40,15 +40,20 @@ function addRandomStock() {
...
-function LoadComments() {
-  const max = getMax();
-  fetch('/data?numCommentsToLoad=' + max).then(response => response.json()).then((comments) => {
-  const commentListElement = document.getElementById('comment-list');
-  commentListElement.innerHTML = ' ';
-  comments.forEach((comment) => {
-  commentListElement.appendChild(createListElement(comment));
+async function LoadComments() {
+  const loggedIn = await isLoggedIn();
+  if(loggedIn) {
```
Comment (dborkan) [inline portfolio/src/main/webapp/script.js]: nit: there should be a space after the `if`

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -40,15 +40,20 @@ function addRandomStock() {
...
+    const max = getMax();
+    fetch('/data?numCommentsToLoad=' + max).then(response => response.json()).then((comments) => {
+    const commentListElement = document.getElementById('comment-list');
+    commentListElement.innerHTML = ' ';
+    comments.forEach((comment) => {
+      commentListElement.appendChild(createListElement(comment));
+      });
     });
-  });
+  }else{
```
Comment (dborkan) [inline portfolio/src/main/webapp/script.js]: nit: also spaces around the else, so `} else {`

**Review by dborkan — COMMENTED** [review]
The new logic looks good, you'll just want to get the updated code from #12 in this branch as well

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -67,3 +72,19 @@ function getMax() {
...
+
+function getMax() {
+  const maxElement = document.getElementById("maxwant");
+  const max = maxElement.options[maxElement.selectedIndex].value;
+  return max;
+}
+
+function deleteComment() {
+  fetch("/delete-list", {method: 'POST'})
+}
```
Comment (dslucas) [inline portfolio/src/main/webapp/script.js]: nit: Seems like this is a duplicate of the code in lines 72-74?

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -40,15 +40,20 @@ function addRandomStock() {
...
 
 // Fetches comments from the server and adds them to the DOM.
-function LoadComments() {
-  const max = getMax();
-  fetch('/data?numCommentsToLoad=' + max).then(response => response.json()).then((comments) => {
-  const commentListElement = document.getElementById('comment-list');
-  commentListElement.innerHTML = ' ';
-  comments.forEach((comment) => {
-  commentListElement.appendChild(createListElement(comment));
+async function LoadComments() {
```
Comment (dslucas) [inline portfolio/src/main/webapp/script.js]:
```suggestion
async function loadComments() {
```

We typically use lower camel case for method names: https://google.github.io/styleguide/jsguide.html#naming-method-names

**Review by dslucas — APPROVED** [review]

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (olu144) [reply portfolio/src/main/webapp/script.js]: addressed

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (olu144) [reply portfolio/src/main/webapp/script.js]: addressed

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (olu144) [reply portfolio/src/main/webapp/script.js]: addressed

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (olu144) [reply portfolio/src/main/webapp/script.js]: addressed

**Pull request merged**